### PR TITLE
adding a dockerfile for aws batch

### PR DESCRIPTION
**Pull request by ramindehghanpoor**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# syntax=docker/dockerfile:experimental
+
+# Use a Python base image with the desired version
```
Comment (shivashankarv) [inline Dockerfile]: I'm wondering if we should just use our base Dockerfile located in CodeCommit?

Comment (shivashankarv) [reply Dockerfile]: That already has mamba, ubuntu etc.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+FROM python:3.9
+#FROM continuumio/miniconda3
+
+# Update and install essential packages
+RUN apt-get update --fix-missing -y && \
```
Comment (shivashankarv) [inline Dockerfile]: You might not need all this.

File: Dockerfile (hunk 1 above)
Comment (shivashankarv) [reply Dockerfile]: Just use the template here maybe: https://github.com/OrnaComputationTeam/mires/blob/main/Dockerfile.mljar
Can be part of a separate issue.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    apt-get clean
+
+# download public key for github.com
+RUN mkdir -p -m 0600 ~/.ssh && ssh-keyscan github.com >> ~/.ssh/known_hosts
+
+# clone spotrna2d repo
+RUN --mount=type=ssh git clone org-80487435@github.com:OrnaComputationTeam/SPOT-RNA-2D.git /SPOT-RNA-2D && cd SPOT-RNA-2D
+
+# install and setup conda3/mamba
+RUN mkdir -p /tmp/conda-build && wget -nv https://repo.anaconda.com/miniconda/Miniconda3-py39_23.3.1-0-Linux-x86_64.sh && bash Miniconda3-py39_23.3.1-0-Linux-x86_64.sh -b -p /miniconda3
```
Comment (shivashankarv) [inline Dockerfile]: All of this is there in the image above.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+# create conda env
+#RUN mamba create -n venv_spotrna_2d python=3.9 && mamba activate venv_spotrna_2d
+
+# install dependencies
+RUN mamba install -y -c bioconda blast=2.14.0
+RUN mamba install -y -c anaconda pandas numpy pathlib
+#RUN mamba install -y -c conda-forge tqdm tensorflow==1.15.0
+
+# install other tools
+RUN mamba install -y -c bioconda viennarna infernal easel
```
Comment (shivashankarv) [inline Dockerfile]: We should have version numbers here.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+# install plmc
+RUN git clone https://github.com/debbiemarkslab/plmc && cd plmc && make all-openmp && cd -
```
Comment (shivashankarv) [inline Dockerfile]: `plmc` is available through conda: https://bioconda.github.io/recipes/plmc/README.html

File: run_rnacmap.sh [hunk 6]
```diff
@@ -1,14 +1,22 @@
 #!/bin/bash
 
+# Start measuring the time
```
Comment (shivashankarv) [inline run_rnacmap.sh]: If need be you can use a `bash` trap here: https://www.notion.so/ornatx/How-to-create-an-EC2-instance-de8df2c7c6054629bdb0fe8d4b04fce7?pvs=4#d39937e26d6c403cb7b24823b7746b97

File: run_rnacmap.sh [hunk 7]
```diff
@@ -1,14 +1,22 @@
 #!/bin/bash
```
Comment (shivashankarv) [inline run_rnacmap.sh:2]: Could you add a #usage comment here. ie on how to run the shell script.

**Review by shivashankarv — CHANGES_REQUESTED** [review]
Few minor comments and suggestions. Will approve if urgent.

File: run_rnacmap.sh (hunk 6 above)
Comment (ramindehghanpoor) [reply run_rnacmap.sh]: Why do we need `bash` trap here? We want to use this shell script in AWS batch.

File: Dockerfile (hunk 4 above)
Comment (ramindehghanpoor) [reply Dockerfile]: As you suggested, I'll use the spec-file instead, so it has the version numbers.

**Review by ramindehghanpoor — COMMENTED** [review]
@shivashankarv, I made your requested changes. It's ready for your review.

**Review by shivashankarv — APPROVED** [review]

**Pull request merged**